kybra/compiler/kybra_post_install/src/generate_candid.rs

```diff
@@ -0,0 +1,49 @@
+use crate::{
+    dfx::dfx,
+    error::{create_error_string, error_to_string},
+};
+
+pub fn generate_candid(canister_name: &str, candid_path: &str) -> Result<(), String> {
+    let candid_output = dfx(
+        "canister",
+        "call",
+        &vec![canister_name, "__get_candid_interface_tmp_hack"],
+    )?;
+
+    if !candid_output.status.success() {
+        return Err(create_error_string(&String::from_utf8_lossy(
+            &candid_output.stderr,
+        )));
+    }
+
+    let mut candid = String::from_utf8_lossy(&candid_output.stdout)
+        .to_string()
+        .replace("\\n", "\n")
+        .replace("\\\"", "\"");
+
+    if let Some(index) = candid.find("(") {
+        candid.drain(index..index + 1);
+    }
+
+    if let Some(index) = candid.find("\"") {
+        candid.drain(index..index + 1);
+    }
+
+    if let Some(index) = candid.rfind(")") {
+        candid.drain(index..index + 1);
+    }
+
+    if let Some(index) = candid.rfind(",") {
+        candid.drain(index..index + 1);
+    }
+
+    if let Some(index) = candid.rfind("\"") {
+        candid.drain(index..index + 1);
+    }
```

Comment:
What are these `rfind` and `drain calls doing? Seems like it's stripping out certain characters or something? If you could pull it into a function that is more declarative that would help here.

Reply:
This code will hopefully become obsolete soon, and since it's mutation I'd rather just leave it all in one function for now, I don't think it's too big of a deal.